src/sparse.rs

```diff
@@ -268,15 +269,15 @@ mod utils {
         let data = &mut data[..len];
         buf.clear();
         buf.reserve_exact(len);
-        for i in 0..len {
-            buf.push((indices[i], data[i]));
+        for (i, v) in indices.iter().zip(data.iter()) {
+            buf.push((*i, *v));
         }
 
         buf.sort_unstable_by_key(|x| x.0);
 
-        for (i, &(ind, x)) in buf.iter().enumerate() {
-            indices[i] = ind;
-            data[i] = x;
+        for (i, v, &mut (ind, x)) in izip!(indices, data, buf) {
```

Comment:
We could use
```rust        
for (&(i, x), (ind, v)) in
            buf.iter().zip(indices.iter_mut().zip(data.iter_mut()))
```
to avoid pulling in itertools

Reply:
I agree it might not worth pulling in itertools just for this small improvement.